homeassistant/auth/__init__.py

```diff
@@ -480,6 +499,37 @@ async def async_remove_refresh_token(
         for revoke_callback in callbacks:
             revoke_callback()
 
+    async def _async_remove_expired_refresh_tokens(
+        self, _: datetime | None = None
+    ) -> None:
+        """Remove expired refresh tokens."""
+        now = time.time()
+        for token in self._store.async_get_refresh_tokens()[:]:
+            if (expire_at := token.expire_at) is not None and expire_at <= now:
+                await self.async_remove_refresh_token(token)
```

Comment:
Maybe we should refactor `async_remove_refresh_token` to not be a coro since nothing is being awaited anymore. that this doesn't have to happen in a task either

Reply:
this would also need some adjustments in `homeassistant.component.auth` as well.
there are even more funtions in `AuthStore` which can be refactored, because the do nothing async anymore.
Would like to create a PR afterwards for this

Reply:
https://github.com/home-assistant/core/pull/108670

Sorry to keep cleaning things up from under this PR.  I haven't really dug into this code before and it seemed better to do it now before merging this so we can keep this simpler

Reply:
ah, no problem, feel free and go ahead 👍

Reply:
I've adapted the code to handle the changes

I still need to test this on my production